src/main/java/org/kiwiproject/eureka/EmbeddedEurekaServer.java

```diff
@@ -0,0 +1,101 @@
+package org.kiwiproject.eureka;
+
+import static com.google.common.base.Preconditions.checkState;
+
+import lombok.extern.slf4j.Slf4j;
+import org.eclipse.jetty.server.Connector;
+import org.eclipse.jetty.server.Server;
+import org.eclipse.jetty.server.ServerConnector;
+import org.eclipse.jetty.webapp.WebAppContext;
+import org.kiwiproject.io.KiwiPaths;
+
+/**
+ * Creates, configures, starts and stops an embedded Jetty server running a Eureka Server.
+ */
+@Slf4j
+public class EmbeddedEurekaServer {
+
+    private static final long IDLE_TIMEOUT = 3_600_000L;
+
+    private final Server eurekaServer;
+    private final ServerConnector connector;
+
+    /**
+     * Creates a new EmbeddedEurekaServer allowing Jetty to pick an available port.
+     */
+    public EmbeddedEurekaServer() {
+        this(0);
+    }
+
+    /**
+     * Creates a new EmbeddedEurekaServer with the given port.
+     *
+     * @param port the port to run Jetty on.
+     */
+    public EmbeddedEurekaServer(int port) {
+        // This is needed so Eureka starts up very quickly
+        System.setProperty("eureka.numberRegistrySyncRetries", "0");
+
+        eurekaServer = new Server();
+        connector = new ServerConnector(eurekaServer);
+
+        // Set timeout option to make debugging easier.
+        connector.setIdleTimeout(IDLE_TIMEOUT);
+
+        connector.setPort(port);
+        eurekaServer.setConnectors(new Connector[] { connector });
+
+        var context = new WebAppContext();
+        context.setServer(eurekaServer);
+        context.setContextPath("/");
+        context.setWar(KiwiPaths.pathFromResourceName("eureka-server.war").toString());
+
+        eurekaServer.setHandler(context);
+    }
+
+    /**
+     * Starts the server.
+     */
+    public void start() {
+        try {
+            eurekaServer.start();
+        } catch (Exception e) {
+            LOG.error("Error while starting down Eureka", e);
```

Comment:
I don't generally start things down...

```suggestion
            LOG.error("Error starting Eureka", e);
```

😂